runtime/testy/runtime.py

```diff
@@ -83,11 +83,16 @@ def nonTestModePrint(*args):
           dawn_connected = False
           continue
         elif newBadThing.event == BAD_EVENTS.ENTER_TELEOP and controlState != "teleop":
-          spawnProcess(PROCESS_NAMES.STUDENT_CODE, runStudentCode, testName, maxIter)
+          terminate_process(PROCESS_NAMES.STUDENT_CODE)
+          name = "teleop"
+          if testName:
```

Comment:
i personally prefer the `if ... else ...` for clarity, but that may just be personal preference.